src/Console/ModelsCommand.php

```diff
@@ -492,7 +492,7 @@ protected function getPropertiesFromTable($model)
             if ($this->write_model_magic_where) {
                 $this->setMethod(
                     Str::camel('where_' . $name),
-                    $this->getClassNameInDestinationFile($model, \Illuminate\Database\Eloquent\Builder::class)
+                    $this->getClassNameInDestinationFile($model, get_class($model->newModelQuery()))
```

Comment:
@mfn 

I checked that the new change to use the external builder for the `where` magic methods is not a big one. We just need to use the builder class returned from the (newModelQuery).

Reply:
Yes this seems appropriate. And as you said, usually I'm also a fan of not mixing fix but sometimes it's ok => 👍 
Thank you